AlfrescoApp/View Controllers/Sync View Controller/RealmSyncViewController.m

```diff
@@ -486,4 +466,134 @@ - (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPa
     }
 }
 
+#pragma mark - Notifications methods
+- (void)sessionReceived:(NSNotification *)notification
+{
+    id<AlfrescoSession> session = notification.object;
+    self.session = session;
+    self.documentFolderService = [[AlfrescoDocumentFolderService alloc] initWithSession:self.session];
+    self.didSyncAfterSessionRefresh = NO;
+    self.title = [self listTitle];
+    
+    [self.navigationController popToRootViewControllerAnimated:YES];
+    // Hold off making sync network requests until either the Sites requests have completed, or a timeout period has passed
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(kSyncOnSiteRequestsCompletionTimeout * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        
+        if (!self.didSyncAfterSessionRefresh)
+        {
+            [self loadSyncNodesForFolder:self.parentNode];
+            self.didSyncAfterSessionRefresh = YES;
+        }
+    });
+}
+
+- (void)siteRequestsCompleted:(NSNotification *)notification
+{
+    [self loadSyncNodesForFolder:self.parentNode];
+    self.didSyncAfterSessionRefresh = YES;
+}
+
+- (void)editingDocumentCompleted:(NSNotification *)notification
+{
+    AlfrescoDocument *editedDocument = notification.object;
+    NSString *editedDocumentIdentifier = [Utility nodeRefWithoutVersionID:editedDocument.identifier];
+    
+    NSIndexPath *indexPath = [self indexPathForNodeWithIdentifier:editedDocumentIdentifier inNodeIdentifiers:[self.collectionViewData valueForKeyPath:kVersionSeriesValueKeyPath]];
+    
+    if (indexPath)
+    {
+        [self.collectionViewData replaceObjectAtIndex:indexPath.row withObject:editedDocument];
+        [self.collectionView reloadItemsAtIndexPaths:@[indexPath]];
+    }
+}
+
+- (void)didUpdatePreference:(NSNotification *)notification
+{
+    NSString *preferenceKeyChanged = notification.object;
+    BOOL isCurrentlyOnCellular = [[ConnectivityManager sharedManager] isOnCellular];
+    
+    if ([preferenceKeyChanged isEqualToString:kSettingsSyncOnCellularIdentifier] && isCurrentlyOnCellular)
+    {
+        BOOL shouldSyncOnCellular = [notification.userInfo[kSettingChangedToKey] boolValue];
+        
+        // if changed to no and is syncing, then cancel sync
+        if ([[RealmSyncManager sharedManager] isCurrentlySyncing] && !shouldSyncOnCellular)
+        {
+            [self cancelSync];
+        }
+        else if (shouldSyncOnCellular)
+        {
+            [self loadSyncNodesForFolder:self.parentNode];
+        }
+    }
+}
+
+- (void)nodeAdded:(NSNotification *)notification
+{
+    NSDictionary *infoDictionary = notification.object;
+    AlfrescoFolder *parentFolder = [infoDictionary objectForKey:kAlfrescoNodeAddedOnServerParentFolderKey];
+    
+    if ([parentFolder.identifier isEqualToString:self.parentNode.identifier])
+    {
+        AlfrescoNode *subnode = [infoDictionary objectForKey:kAlfrescoNodeAddedOnServerSubNodeKey];
+        [self addAlfrescoNodes:@[subnode] completion:nil];
```

Comment:
Please avoid the potential "attempt to insert nil object" crash if subnode is nil.